test/distribution/test_distribution_sampling.py

```diff
@@ -299,3 +300,109 @@ def test_inflated_beta_sampling(
         np.histogram(samples.asnumpy())[0],
         rtol=0.08,
     )
+
+
+@pytest.mark.parametrize("rate", [1000.0, 10.0])
+@pytest.mark.parametrize("zero_probability", [0.1, 0.4, 0.9])
+def test_inflated_poisson_sampling(
+    rate: float,
+    zero_probability: float,
+):
+    distr = ZeroInflatedPoissonOutput().distribution(
+        distr_args=[
+            mx.nd.array([[1-zero_probability, zero_probability]]),
+            mx.nd.array([rate]),
+            mx.nd.array([0.]),
+        ],
+    )
+
+    quantiles = [0.2, 0.4, 0.5, 0.6, 0.8]
+    samples = distr.sample()
+    assert samples.shape == (1,)
+    num_samples = 100_000
+    samples = distr.sample(num_samples)
+    assert samples.shape == (num_samples, 1)
+
+    category = np.random.choice(
+        [0, 1],
+        p=[
+            zero_probability,
+            1 - zero_probability,
+        ],
+        size=num_samples,
+    )
+    samples_numpy = np.random.poisson(lam=rate, size=num_samples)
+    samples_numpy[category == 0] = 0
+
+    # mean
+    assert np.allclose(
+        samples.mean().asnumpy(), distr.mean.asnumpy(), atol=1e-2, rtol=1e-2
+    )
+    # var
+    assert np.allclose(
+        np.var(samples.asnumpy()),
+        distr.variance.asnumpy(),
+        atol=1e-2,
+        rtol=1e-2,
+    )
+
+    # quantiles
+    assert np.allclose(
+        np.quantile(samples_numpy, q=quantiles),
+        np.quantile(samples.asnumpy(), q=quantiles),
+        rtol=2,
+    )
+
+
+# @pytest.mark.parametrize("mu_alpha", [(5.0, 0.5)])
+# @pytest.mark.parametrize("zero_probability", [0.0, 0.3, 0.9])
+# def test_inflated_neg_binomial_sampling(
+#     mu_alpha: float,
+#     zero_probability: float,
+# ):
+#     mu, alpha = mu_alpha
```

Comment:
@kaijennissen is this test meant to run? If not, it's maybe better to decomment it and `@pytest.mark.skip(...)`-it

Other than that, the PR looks good to me :+1:

Reply:
@lostella I was not sure whether this test is necessary or not since the `ZeroInflatedNegativeBinomialOutput` is a subclass of `MixtureDistributionOutput` which has its own tests. What do you think?

Reply:
I think it's fine not to have it, but then maybe you could just remove the commented lines (just to avoid confusion, no big deal)

Reply:
Removed.